documentation/creating_a_script_plan.md

```diff
@@ -0,0 +1,369 @@
+   This command creates a Bolt YAML plan that takes a `targets` parameter, runs the script on the
+   targets, and returns the result from the script run. If you'd prefer to create a Puppet language
+   plan instead of YAML, you can run the same command with the `--pp` option.
+
+   _\*nix shell command_
+   ```
+   bolt plan new manage_docker::update_images --script manage_docker/scripts/update_images.sh
+   ```
+   _PowerShell cmdlet_
+   ```powershell
+   New-BoltPlan -Name manage_docker::update_images -Script manage_docker/scripts/update_images.sh
+   ```
+
+Et voila! Your plan is now in `plans/update_images.yaml` in your project. You can now run your plan
+with:
+
+_\*nix shell command_
+```
+bolt plan run manage_docker::update_images -t <targets>
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltPlan -Name manage_docker::update_images -Targets <targets>
+```
+
+## Scripts with arguments
+
+### As environment variables
+
+You can pass arguments from your plan to your script as environment variables, which can be useful
+for scripts with optional parameters (where ordering arguments can be hard to predict), complex
+arguments, or structured data.
+
+Modify the example script above to take a command-line argument for the directory to run
+`docker-compose` from:
+
+_update\_images.sh_
+```
+#!/bin/sh
+if [ -n $DOCKER_COMPOSE_DIRECTORY ]
+  cd $DOCKER_COMPOSE_DIRECTORY
+fi
+for c in `docker-compose ps --services |sort`; do
+  echo "redoing $c"
+  docker-compose rm -s -f $c
+  docker-compose pull $c
+done
+docker-compose up -d --remove-orphans
+```
+
+Run this script directly with Bolt by passing the `--env-var` command-line option to set
+environment variables during script execution:
+
+_\*nix shell command_
+```
+bolt script run manage_docker/scripts/update_images.sh -t <targets> --env-var DOCKER_COMPOSE_DIRECTORY=./docker
+```
+
+_PowerShell cmdlet_
+```powershell
+Invoke-BoltScript -Name manage_docker/scripts/update_images.sh -Targets <targets> -EnvVar DOCKER_COMPOSE_DIRECTORY=.\docker
+```
+
+Before you pass the argument through the plan as an environment variable,
+[add a plan parameter](writing_yaml_plans.md#parameters) for the argument and pass it through to the
+script. In a YAML plan, that should look something like this:
+```yaml
+parameters:
+  directory:
+    type: Optional[String]
+    default: .
+    description: The directory to execute 'docker-compose' from
+  targets:
+    ...
+```
+
+Next set that argument as an environment variable as part of the [script
+step](writing_yaml_plans.md#script_step)
+```yaml
+steps:
+  - name: run_script
+    script: manage_docker/scripts/update_images.sh
+    env_vars:
+      DOCKER_COMPOSE_DIRECTORY: ./docker
+    targets: $targets
+```
+
+And now your plan should look something like this:
+```yaml
+description: Update Docker images
+parameters:
+  directory:
+    type: Optional[String]
+    default: .
+    description: The directory to execute 'docker-compose' from
```

Comment:
This parameter is never used. Is it supposed to set `env_vars.DOCKER_COMPOSE_DIRECTORY` ?

Reply:
Oo, yes my bad.